Directory.Build.props

```diff
@@ -45,6 +45,13 @@
     <NoWarn Condition="'$(Language)' == 'C#'">$(NoWarn),1573,1591,1712</NoWarn>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
+    <!-- When building in source build mode, treat this set of warnings not as errors. Source build is
+         using a newer version of Microsoft.CodeAnalysis than the default build -->
+    <NoWarn>$(NoWarn);CS8600;CS0618;CS8601;CS8602;CS8603;CS8604</NoWarn>
```

Comment:
There are thousands of CA warnings when the version of Microsoft.CodeAnalysis is upgraded. These are set to nowarn in full source-build mode. Not sure whether it's worth taking a look at these.